lib/matplotlib/dates.py

```diff
@@ -207,16 +207,19 @@ def _get_rc_timezone():
 
 def _to_ordinalf(dt):
     """
-    Convert :mod:`datetime` or :mod:`date` to the Gregorian date as UTC float
-    days, preserving hours, minutes, seconds and microseconds.  Return value
-    is a :func:`float`.
+    Convert :mod:`datetime`, :mod:`date` or :mod:`timedelta` to the Gregorian
+    date as UTC float days, preserving hours, minutes, seconds and
+    microseconds. Return value is a :func:`float`.
     """
     # Convert to UTC
     tzi = getattr(dt, 'tzinfo', None)
     if tzi is not None:
         dt = dt.astimezone(UTC)
         tzi = UTC
 
+    if isinstance(dt, datetime.timedelta):
```

Comment:
how else would you compute widths? It is a similar problem to temperatures. You can't simply convert a 5 degrees C temperature change into Fahrenheit temperature change (a big annoyance of mine when reading news articles that blindly injects unit conversions)